File: sicp/1-building-abstractions-with-procedures/1-2/1-2-1/1.py
def factorial(n):
    if n == 1:
        return 1
    else:
        return n * factorial(n-1)

print(factorial(5)) #120

def factorial2(n):
    return fact_iter(1, 1, n)

def fact_iter(product, counter, maxn):
    if counter > maxn:
        return product
    else:
        return fact_iter(product*counter, counter+1, maxn)

print(factorial2(6)) #720

File: sicp/1-building-abstractions-with-procedures/1-2/1-2-1/1.php
<?php

function factorial($n) {
    if ($n === 1) {
        return 1;
    } else {
        return $n * factorial($n - 1);
    }
}

echo factorial(5),"\n";

function factorial2($n) {
    return factIter(1, 1, $n);
}

function factIter($product, $counter, $maxn) {
    if ($counter > $maxn) {
        return $product;
    } else {
        return factIter($product * $counter, $maxn);
    }
}

echo factorial(6),"\n";